### ISSUE TITLE
Episode detection can't recognise iTunes naming

### ISSUE THREAD

**Issue by Deddiekoel** [SiCKRAGETV/sickrage-issues#581]
Episodes that are stored in the iTunes naming convention (Season %S/%0E %SN) aren't recognised by Sickrage. When scanning the folders it gives the following error:

2015-02-10 18:38:53 SHOWQUEUE-REFRESH :: No episode number found in /.../TV Shows/Gotham/Season 1/14 The Fearsome Dr. Crane.mp4, ignoring it

Also, it won't accept Season %S/%0E %SN as a custom naming convention. Which might be related.

**Comment by Deddiekoel** [comment]
I have only recently switched to Sickrage and with other Sickbeard versions I never had this issue. So I compared the code with the last Sickbeard version I used and I found the difference in the regexes.py file.
Sickrage's file contains this for the "no season" regex:

    ('no_season',
     # Show Name - 01 - Ep Name
     # 01 - Ep Name
     # 01 - Ep Name
     '''
     ^((?P<series_name>.+?)(?:[. _-]{2,}|[. _]))?    # Show_Name and separator
     (?P<ep_num>\d{1,3})                             # 02
     (?:-(?P<extra_ep_num>\d{1,3}))*                 # -03-04-05 etc
     \s?of?\s?\d{1,3}?                               # of joiner (with or without spaces) and series total ep
     [. _-]+((?P<extra_info>.+?)                     # Source_Quality_Etc-
     ((?<![. _-])(?<!WEB)                            # Make sure this is really the release group
     -(?P<release_group>[^- ]+([. _-]\[.*\])?))?)?$  # Group
     '''
    ),

And an older, working, version contained:

              ('no_season',
               # Show Name - 01 - Ep Name
               # 01 - Ep Name
               # 01 - Ep Name
               '''
               ^((?P<series_name>.+?)(?:[. _-]{2,}|[. _]))?             # Show_Name and separator
               (?P<ep_num>\d{1,2})                           # 02
               (?:-(?P<extra_ep_num>\d{1,2}))*               # 02
               [. _-]+((?P<extra_info>.+?)                 # Source_Quality_Etc-
               ((?<![. _-])(?<!WEB)                        # Make sure this is really the release group
               -(?P<release_group>[^- ]+))?)?$              # Group
               '''
               ),

The difference is this line (which has been added in the Sickrage version):

     \s?of?\s?\d{1,3}?                               # of joiner (with or without spaces) and series total ep

When I remove this line everything is my folder is picked up again.

**Comment by fernandog** [comment]
I was added for a reason. Maybe SR can add another one instead of change this one.

**Comment by Deddiekoel** [comment]
I added the "old" version under the name 'itunes' (for lack of a better name). Removed a show from the DB and added it again. All episodes where recognised. This seems like a quick fix at least.

**Comment by fernandog** [comment]
@Deddiekoel can you do that "edit" here and make a Pull Request/?

https://github.com/SiCKRAGETV/SickRage/edit/develop/sickbeard/name_parser/regexes.py

**Comment by Deddiekoel** [comment]
Yup, willdo!

**Comment by abeloin** [comment]
For reference: the commit introducing this change: https://github.com/SiCKRAGETV/SickRage/commit/88a3444db2b65ad3cc6f86a5a65c920a09a7ce2c

**Comment by fernandog** [comment]
@Deddiekoel did you created the PULL REQUEST? Nothing appeared

**Comment by Deddiekoel** [comment]
I thought I did, let me try again...

**Comment by Deddiekoel** [comment]
Created a fork but forgot to press the pull request button. Did so now. Did you see it?

**Comment by fernandog** [comment]
yes. have you fully tested?

**Comment by Deddiekoel** [comment]
Limited, it works on my implementation (on a Synology NAS). 
However, what I've added is the code from a different Sickbeard build (which was hopefully tested). Also I have only added a new regex and haven't touched anything else.

**Comment by fernandog** [comment]
@Deddiekoel develop updated. please test and report back

**Comment by abeloin** [comment]
Reverted the commit as it was catching non itunes file. E.g.: constantine.113.hdtv-lol.mp4

**Comment by abeloin** [comment]
@Deddiekoel

**Comment by fernandog** [comment]
@Deddiekoel please reply, otherwise will close

**Comment by Deddiekoel** [comment]
I'm redoing the regex to be completely iTunes specific. I'll submit a new pull request for it. 

Verstuurd vanaf mijn iPhone

**Comment by fernandog** [comment]
@Deddiekoel any luck?

**Comment by Deddiekoel** [comment]
Almost there. One show has some issues. 

Verstuurd vanaf mijn iPhone

**Comment by fernandog** [comment]
Remember that some uploaders dont use the correct name pattern. A few may be left out

**Comment by Deddiekoel** [comment]
Yeah, but this regex is for people who use iTunes to manage their collection. 
The convention is stupid simple. But some shows just aren't picked up...

Verstuurd vanaf mijn iPhone

**Comment by fernandog** [comment]
Post your regex and the test string (filenames). Will try to help you

**Comment by Deddiekoel** [comment]
Ok, the convention iTunes uses is dirt simple:

[Show Name]\[Season folder]\[Ep Number] [Episode Name]

This is re regex I have now:

    ('itunes',
     # 01 Ep Name
     '''
     ^(?P<ep_num>\d{1,3})[ ](?P<extra_info>.*)?$      # 02 Episode Name
     '''),

That seems to work, but only if I put it in as the first entry. Otherwise SR uses the season_only regex:

2015-02-22 21:19:04 ERROR    SHOWQUEUE-REFRESH :: No episode number found in /volume1/media/TV Shows/Broadchurch/Season 2/07 Episode 7.mp4, ignoring it
2015-02-22 21:19:04 INFO     SHOWQUEUE-REFRESH :: parse_result: S22015-07-07 [ANIME VER: -1] [ABD: True] [ANIME: False] [whichReg: ['scene_date_format', 'season_only']]

**Comment by Deddiekoel** [comment]
Just submitted a pull request with the above change!

**Comment by fernandog** [comment]
https://github.com/SiCKRAGETV/SickRage/pull/1407

**Comment by fernandog** [comment]
@Deddiekoel we are waiting your reply here 
https://github.com/SiCKRAGETV/SickRage/pull/1407

otherwise we will close this
